doc/release-process.md

```diff
@@ -12,10 +12,34 @@ It is best if the maintainers are present during the release, so they can help e
 
 This process also assumes that there will be no minor releases for old major releases.
 
+We aim to cut a regular release every 3-4 months, approximately twice as frequent as major Bitcoin Core releases. Every second release should be published one month before the feature freeze of the next major Bitcoin Core release, allowing sufficient time to update the library in Core.
+
+## Sanity Checks
+Perform these checks before creating a release:
+
+1. Ensure `make distcheck` doesn't fail.
+```shell
+make distcheck
+```
+2. Check installation with autotools:
+```shell
+dir=$(mktemp -d)
+./configure --prefix=$dir && make install && ls -l $dir/include $dir/lib
+gcc -o ecdsa examples/ecdsa.c -I $dir/include -L $dir/lib/ -l secp256k1 && ./ecdsa
+```
+3. Check installation with CMake:
+```shell
+dir=$(mktemp -d)
+mkdir build && cmake -B build -DCMAKE_INSTALL_PREFIX=$dir && cmake --build build --target install && ls -l $dir/include $dir/lib*
+gcc -o ecdsa examples/ecdsa.c -I $dir/include -L $dir/lib*/ -l secp256k1 && ./ecdsa
+```
+
 ## Regular release
 
 1. Open a PR to the master branch with a commit (using message `"release: prepare for $MAJOR.$MINOR.$PATCH"`, for example) that
-   * finalizes the release notes in [CHANGELOG.md](../CHANGELOG.md) (make sure to include an entry for `### ABI Compatibility`),
+   * finalizes the release notes in [CHANGELOG.md](../CHANGELOG.md) by
+       * adding a section for the release (make sure that the version number is a link to a diff between the previous and new version) and
+       * including an entry for `### ABI Compatibility` if it doesn't exist,
```

Comment:
Speaking of ABI compatibility, does it make sense to specify SONAME on Linux, for example `libsecp256k1.so.2`.

Reply:
You mean, saying that the shared library is `libsecp256k1.so.2`? I like this idea. It makes the ABI version very explicit.

Then we probably want to say bikeshed a bit about the exact phrasing; Does this apply to macOS, too? Do we want to say something for Windows? Etc... (So that should not hold up 0.3.2 now, I think.)

Reply:
> You mean, saying that the shared library is `libsecp256k1.so.2`?

To be pedantic, the shared library's [soname](https://tldp.org/HOWTO/Program-Library-HOWTO/shared-libraries.html) is `libsecp256k1.so.2`.

> I like this idea. It makes the ABI version very explicit.

That is my point :)

> Then we probably want to say bikeshed a bit about the exact phrasing; Does this apply to macOS, too? Do we want to say something for Windows? Etc... (So that should not hold up 0.3.2 now, I think.)

I'd skip them for now. While an ABI version is a part of DYLIB and DLL names, AFAIK, linking processes on macOS and Windows do not rely on it.